src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -1276,6 +1276,10 @@ bool Pet::addSpell(uint32 spellId, ActiveStates active /*= ACT_DECIDE*/, PetSpel
         return false;
     }
 
+    // SPELL_ATTR4_UNK15 = DO NOT ADD SPELL TO SPELLBOOK OR ACTIONBAR
```

Comment:
`SPELL_ATTR4_UNK15` 

Is used to prevent server to teach someone those spells.
These spells are used serversidely only to fix the behavior of an attack.